File: src/main/java/Library/Catalogue.java
package Library;

import java.util.ArrayList;
import java.util.List;

//  Catalogue: Handing the actions about books
public class Catalogue {
    ArrayList<Book> catalogue;

    // Initializing
    public Catalogue(){
        catalogue = new ArrayList<Book>();
    }

    // Display the books
    public List<Book> allBooks(){
        return catalogue;
    }

    // adders
    public void addBook(Book book){
        catalogue.add(book);
    }

    // getters
    public int getCatalogueSize(){
        return catalogue.size();
    }
    public Book getBook(int index){
        return catalogue.get(index);
    }
    public Book getBook(String title) {
        for (Book b : catalogue) {
            if (b.getTitle().equalsIgnoreCase(title)) {
                return b;
            }
        }
        return null;
    }

}

File: A3/models/Catalogue.js
// Catalogue.js

class Catalogue {
    constructor() {
        this.catalogue = [];   // Array of Book objects
    }

    // Return full list
    allBooks() {
        return this.catalogue;
    }

    // Add book
    addBook(book) {
        this.catalogue.push(book);
    }

    // Return catalogue size
    getCatalogueSize() {
        return this.catalogue.length;
    }

    // Get book by index
    getBookByIndex(index) {
        return this.catalogue[index];  // Java behavior: no bounds check
    }

    // Get book by title
    getBookByTitle(title) {
        for (const b of this.catalogue) {
            if (b.getTitle().toLowerCase() === title.toLowerCase()) {
                return b;
            }
        }
        return null;
    }

    // --- Combined getBook(param) to match Java overloading ---
    getBook(param) {
        if (typeof param === "number") {
            return this.getBookByIndex(param);
        }
        if (typeof param === "string") {
            return this.getBookByTitle(param);
        }
        return null;
    }
}

module.exports = Catalogue;
